src/App.tsx

```diff
@@ -16,47 +16,93 @@ export const goodsFromServer = [
 ];
 
 export const App: React.FC = () => {
+  const [
+    sortedGoods,
+    setSortedGoods,
+  ] = useState(goodsFromServer);
+  const [isVisibleReser, setIsVisibleReser] = useState(false);
+
+  enum SortType {
+    Alphabet = 'alphabet',
+    Length = 'length',
+    Reverse = 'reverse',
+  }
```

Comment:
Things that do not directly affect the component should be taken outside the definition of the component, because they will be created anew each time after each rerender.